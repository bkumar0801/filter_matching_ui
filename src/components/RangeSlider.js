import React from 'react';
import '../style/App.css';

class RangeSlider extends React.Component {
  
    constructor(props) {
        super(props)
        this.state = { value: 18 }
    }
    
    changeValue(e) {
      this.setState({
        value: e.currentTarget.value
      });
      this.props.callbackParent(e.currentTarget.value);
    }
    
    render() {
      return (
        <div>
          <input type="range" name="quantity" min="18" max="95" onInput={(e)=>this.changeValue(e)} value={this.state.value} />
          <output for="quantity">{this.state.value}</output>
        </div>
      );
    }
  };

  export default RangeSlider;
  