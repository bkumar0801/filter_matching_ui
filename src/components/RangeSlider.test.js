import React from 'react';
import { shallow, enzyme } from 'enzyme';
import RangeSlider from './RangeSlider';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {sinon, spy, stub} from 'sinon';

configure({ adapter: new Adapter() });

describe(RangeSlider, () => {
    const mockOnInput = jest.fn();
    const component = shallow(<RangeSlider initialValue = {20}  min = {18} max = {95} onInput={mockOnInput} />); 

    it('renders and matches our snapshot', () => {
        const component = renderer.create(<RangeSlider />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("contains input and output", () => {
        expect(component.find("input")).toHaveLength(1);
        expect(component.find("output")).toHaveLength(1);
    });

    it("has state set", () => {
        expect(component.state('value')).toEqual(20);
        expect(component.state('min')).toEqual(18);
        expect(component.state('max')).toEqual(95);
    });

    it('change value of range on setState', () => {
        const wrapper = shallow(<RangeSlider initialValue = {20}  min = {18} max = {95}/>);
        let range = wrapper.find({ type: 'range' });
        expect(range.props().value).toEqual(20);
        wrapper.setState({ value: 40 });
        range = wrapper.find({ type: 'range' });
        expect(range.props().value).toEqual(40);
    });
});