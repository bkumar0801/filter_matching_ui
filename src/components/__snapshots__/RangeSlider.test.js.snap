// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class RangeSlider extends _react2.default.Component {

  constructor(props) {
    super(props);
    this.state = { value: props.initialValue, min: props.min, max: props.max };
  }

  onChange(e) {
    this.setState({
      value: e.currentTarget.value });

    this.props.callbackParent(e.currentTarget.value);
  }

  render() {
    return (
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 20 }, __self: this },
        _react2.default.createElement('input', { type: 'range', name: 'quantity', min: this.state.min, max: this.state.max, onChange: e => this.onChange(e), value: this.state.value, __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }),
        _react2.default.createElement('output', { htmlFor: 'quantity', __source: { fileName: _jsxFileName, lineNumber: 22 }, __self: this }, this.state.value)));


  }} renders and matches our snapshot 1`] = `
<div>
  <input
    max={undefined}
    min={undefined}
    name="quantity"
    onChange={[Function]}
    type="range"
    value={undefined}
  />
  <output
    htmlFor="quantity"
  />
</div>
`;
